File: integration-test/java/quarkus-app/src/main/java/de/qualityminds/lazyval/integration/Startup.java
package de.qualityminds.lazyval.integration;

import de.qualityminds.lazyval.integration.shared.Isbn;
import de.qualityminds.lazyval.integration.shared.Quantity;
import io.quarkus.runtime.StartupEvent;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.event.Observes;
import jakarta.transaction.Transactional;

@ApplicationScoped
public class Startup {

    @Transactional
    void onStart(@Observes StartupEvent ev) {
        Order orderA = new Order(Isbn.parse("3-86680-192-0"), new Quantity(1), new EMail("a@b.de"));
        Order orderB = new Order(Isbn.parse("978-3-86680-192-9"), new Quantity(1), new EMail("x@y.de"));

        orderA.persist();
        orderB.persist();
    }
}

File: integration-test/kotlin/spring-app/src/main/kotlin/de/qualityminds/lazyval/integration/Startup.kt
package de.qualityminds.lazyval.integration

import de.qualityminds.lazyval.integration.shared.Isbn
import de.qualityminds.lazyval.integration.shared.Quantity
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
class Startup(private val orderRepository: OrderRepository) {

    @EventListener(ApplicationReadyEvent::class)
    @Transactional
    fun onStart() {
        val orderA = Order(
            isbn = Isbn.parse("3-86680-192-0"),
            quantity = Quantity(1),
            email = EMail("a@b.de")
        )
        val orderB = Order(
            isbn = Isbn.parse("978-3-86680-192-9"),
            quantity = Quantity(1),
            email = EMail("x@y.de")
        )

        orderRepository.save(orderA)
        orderRepository.save(orderB)
    }
}